README.md

```diff
@@ -146,3 +146,27 @@ stable API, we also improve the API over time. As a result, we typically
 deprecate methods before removing them, according to the [deprecation guidelines].
 
 [deprecation guidelines]: https://datafusion.apache.org/library-user-guide/api-health.html
+
+## Dependencies and a `Cargo.lock`
```

Comment:
@mbrobbel  has a good point here https://github.com/apache/datafusion/pull/14069#issuecomment-2582533271

I think one concern originally was how we would keep a `Cargo.lock` file up to date with the latest version of the dependencies

I can't remember if depndabot was available at that point -- maybe dependabot is good enough that we could check in Cargo.lock and have Dependabot update it

Reply:
I'd like to point to https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html for considerations and suggestions.

Reply:
TLDR it sounds like the rust team now suggests always committing `Cargo.lock` and letting dependabot handle updates. That seems like a good idea to me

Reply:
One thing we have to be aware of in DataFusion is that as part of the Apache security posture, only certain third party actions are allowed -- we would have to double check Rennovate

I think the next step is probably to file an issue to explicitly discuss checking in a Cargo.lock file. I'll try and find time over the next few days

Reply:
I filed a ticket to discuss next steps:
- https://github.com/apache/datafusion/issues/14135